sandbox/docker_entrypoint.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+# Runs the command provided as arguments, but attempts to configure permissions first.
+
+important_read_dirs=("/grist" "/persist")
+important_write_dirs=("/persist")
+current_user_id=$(id -u)
+
+# We want to avoid running Grist as root if possible.
+# Try to setup permissions and de-elevate to a normal user.
+if [[ $current_user_id == 0 ]]; then
+  target_user=${GRIST_DOCKER_USER:-grist}
+  target_group=${GRIST_DOCKER_GROUP:-grist}
+
+  for dir in "${important_write_dirs[@]}"; do
+    # Make sure the target user owns everything that Grist needs write access to.
+    find "$dir" ! -user "$target_user" -exec chown "$target_user" "{}" +
+  done
+
+  # Restart as the target user, replacing the current process (replacement is needed for security).
+  # Alternative tools to setpriv are: chroot, gosu.
+  exec setpriv --reuid "$target_user" --regid "$target_group" --init-groups /usr/bin/env bash "$BASH_SOURCE" "$@"
```

Comment:
Although a valid bashism, I find `$BASH_SOURCE` a little obscure here, as you're relying on bash's behaviour of `$lol` being a synonym for `${lol[0]}` for an array `lol`.

Can we just call it `$0` instead, as we do with other shell scripts in Grist?

Reply:
Can't remember why I chose one over the other - happy to move to $0